### DISCUSSION TITLE
About setting custom workspace location configuration in config file availability.

### DISCUSSION THREAD

**Discussion by Kambaa** [dbeaver/dbeaver#15926] [Q&A]
Hello, 

Instead of giving it as an argument of the dbeaver.exe, is there a possible way to set the [custom location](https://github.com/dbeaver/dbeaver/wiki/Workspace-Location#custom-location) setting in a configuration file?( in dbeaver.ini or configuration/configuration.ini) 

Installing and setting up an application is problem for me (You can imagine the cases, using work computers, a.d./exchange limitations/restrictions, disappearance of folders in %appdata% at regular intervals, in short, portable usage is more comfortable for me, portable'em all the way!)

Thank you.

**Comment by Matvey16** [comment]
@Kambaa Hello, thank you for a suggestion. Unfortunately the .ini file is read only after the workspace has already been initialized and that process is a part of eclipse platform, so we don't have means to change that.

**Comment by Kyogre** [comment]
For now workarounds are:
1. You can create a .bat file (name it like "dbeaver launch.bat") inside a folder where your dbeaver.exe is located. In the .bat file write this code to make workspace directory be stored inside dbeaver's folder:
`start "%~dp0" "dbeaver.exe" -data workspace`
Then you can even create shortcut to that .bat file and set an icon for it.

2. You can use any "Bat To Exe Converter" software to make .exe out of the .bat from workaround №1.

**Reply by nklamann** [reply]
I used brute force and created a symlink: 
`\AppData\Roaming>  mklink /J DBeaverData <wherever>`

**Comment by seidnerj** [comment]
For anyone looking to do something like this on macOS, here's my take:

1. In /Applications folder, copy "DBeaver.app" to "_DBeaver.app".
2. Right click on DBeaver -> "Show Package Contents".
3. Delete everything under the Contents folder apart from "Info.plist", "Resources" & "MacOS".
4. In the MacOS folder, delete the "dbeaver" executable and replace it with a shell script file with the same name that has the following content (obviously, modify the "-data" parameter as you see fit):

```
#!/usr/bin/env bash
/Applications/_DBeaver.app/Contents/MacOS/dbeaver -data ~/Documents/DBeaverData/workspace6/
```

5. Make sure the script is executable by running "chmod +x dbeaver" from a terminal prompt.
